sdk/tests/tutorials/ibor/test_reference_portfolios.py

```diff
@@ -33,16 +34,26 @@ def setUpClass(cls):
         cls.instrument_loader = InstrumentLoader(cls.instruments_api)
         cls.instrument_ids = cls.instrument_loader.load_instruments()
 
+        cls.id_generator = IdGenerator(default_scope=TestDataUtilities.tutorials_scope)
+
+    @classmethod
+    def tearDownClass(cls):
+        for _, scope, code in cls.id_generator.pop_scope_and_codes():
```

Comment:
This generator method is brilliant in how it hides away the need to pop id's off the array from the user :) nice stuff